distributed/tests/test_cancelled_state.py

```diff
@@ -561,3 +567,127 @@ def test_resume_executing_worker_state(ws_with_running_task):
     assert not instructions
     assert ws.tasks["x"] is ts
     assert ts.state == prev_state
+
+
+def test_worker_state_flight_to_flight(ws):
+    """Test state loop:
+
+    flight -> cancelled -> fetch
+
+    Test that the task immediately reverts to its original state.
+    """
+    ws2 = "127.0.0.1:2"
+    instructions = ws.handle_stimulus(
+        ComputeTaskEvent.dummy("y", who_has={"x": [ws2]}, stimulus_id="s1"),
+        FreeKeysEvent(keys=["y", "x"], stimulus_id="s2"),
+        ComputeTaskEvent.dummy("z", who_has={"x": [ws2]}, stimulus_id="s3"),
+    )
+    assert instructions == [
+        GatherDep(worker=ws2, to_gather={"x"}, total_nbytes=1, stimulus_id="s1")
+    ]
+    assert ws.tasks["x"].state == "flight"
+
+
+def test_worker_state_executing_success_to_fetch(ws_with_running_task):
+    """Test state loops:
+
+    - executing -> cancelled -> resumed -> fetch
+    - executing -> long-running -> cancelled -> resumed -> fetch
```

Comment:
```suggestion
    - executing -> cancelled -> resumed -> memory
    - executing -> long-running -> cancelled -> resumed -> memory
```
since we should never actually enter the fetch state to my understanding

Reply:
ty